kernel/src/memory/buddy_allocator.rs

```diff
@@ -190,49 +190,28 @@ macro_rules! buddy_allocator_bitmap_tree {
                 let index = level_offset + ((ptr as usize) >> (order + $BASE_ORDER)) + 1;
 
                 assert!(index < BLOCKS_IN_TREE, "Block index {} out of bounds!", index);
+
                 assert_eq!(
                     unsafe { self.block(index - 1).order_free },
                     0,
                     "Block to free (index {}) must be used!",
                     index,
                 );
 
-                // Only if order isn't 0 we need to check the children, as blocks of order 0 have
```

Comment:
Why was this removed, btw?

Reply:
The whole algorithm was buggy... idk. It works now and all tests pass lol.